pkg/api/tag_test.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/ethersphere/bee/pkg/api"
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/jsonhttp/jsonhttptest"
+	mp "github.com/ethersphere/bee/pkg/pusher/mock"
+	"github.com/ethersphere/bee/pkg/storage/mock"
+	"github.com/ethersphere/bee/pkg/storage/mock/validator"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/tags"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+func TestTags(t *testing.T) {
+	var (
+		resource        = func(addr swarm.Address) string { return "/bzz-chunk/" + addr.String() }
+		tagResource     = func(addr swarm.Address) string { return "/bzz-tag/addr/" + addr.String() }
+		tagResourceUUid = func(uuid uint64) string { return "/bzz-tag/uuid/" + strconv.FormatUint(uuid, 10) }
+		validHash       = swarm.MustParseHexAddress("aabbcc")
+		validContent    = []byte("bbaatt")
+
+		mockValidator        = validator.NewMockValidator(validHash, validContent)
+		tag                  = tags.NewTags()
+		mockValidatingStorer = mock.NewValidatingStorer(mockValidator, tag)
+		mockPusher           = mp.NewMockPusher(tag)
+		client               = newTestServer(t, testServerOptions{
+			Storer: mockValidatingStorer,
+			Tags:   tag,
+		})
+	)
+
+	t.Run("tag-header-in-return", func(t *testing.T) {
+		headers := make(map[string][]string)
+		headers[api.TagHeaderName] = []string{"2341312131"} // the value doesn't matter, it is random so we cant test for that
+		rcvdHeaders := jsonhttptest.ResponseDirectReceiveHeaders(t, client, http.MethodPost, resource(validHash), bytes.NewReader(validContent), http.StatusOK, jsonhttp.StatusResponse{
+			Message: http.StatusText(http.StatusOK),
+			Code:    http.StatusOK,
+		})
+
+		found := false
+		for k, v := range rcvdHeaders {
+			if api.TagHeaderName == strings.ToLower(k) {
+				found = true
+				_, err := strconv.ParseUint(v[0], 10, 32)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+		}
+
+		if !found {
+			t.Fatalf("could not find tagid header in response")
+		}
```

Comment:
Can this be replaced with `if rcvdHeaders.Get(api.TagHeaderName) != uid { t.Fatalf(...) }`?